omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -260,38 +184,17 @@ where
             && self.state.1 == Phase::Accept
             && self.handle_sequence_num(dec.seq_num, dec.n.pid) == MessageStatus::Expected
         {
+            if dec.decided_idx > self.internal_storage.get_accepted_idx() {
+                let new_accepted_idx = self.internal_storage.flush_batch().expect(WRITE_ERROR_MSG);
+                self.handle_flushed_accepted(dec.n, new_accepted_idx);
+            }
             self.internal_storage
                 .set_decided_idx(dec.decided_idx)
-                .expect("storage error while trying to write decided index");
-        }
-    }
-
-    fn accept_entries_follower(&mut self, n: Ballot, entries: Vec<T>) -> StorageResult<()> {
-        let accepted_res = self
-            .internal_storage
-            .append_entries_and_get_accepted_idx(entries)?;
-        if let Some(accepted_idx) = accepted_res {
-            self.handle_flushed_accepted(n, accepted_idx);
+                .expect(WRITE_ERROR_MSG);
         }
-        Ok(())
     }
 
-    #[cfg(feature = "unicache")]
-    fn accept_encoded_entries_follower(
-        &mut self,
-        n: Ballot,
-        encoded_entries: Vec<T::EncodeResult>,
-    ) -> StorageResult<()> {
-        let accepted_res = self
-            .internal_storage
-            .append_encoded_entries_and_get_accepted_idx(encoded_entries)?;
-        if let Some(accepted_idx) = accepted_res {
-            self.handle_flushed_accepted(n, accepted_idx);
-        }
-        Ok(())
-    }
-
-    fn handle_flushed_accepted(&mut self, n: Ballot, accepted_idx: u64) {
+    fn handle_flushed_accepted(&mut self, n: Ballot, accepted_idx: usize) {
```

Comment:
To make code more intuitive when reading the message handlers, we can rename this

```suggestion
    fn reply_accepted(&mut self, n: Ballot, accepted_idx: usize) {
```